pdns/lwres.cc

```diff
@@ -389,7 +403,7 @@ LWResult::Result asyncresolve(const ComboAddress& ip, const DNSName& domain, int
 
 #ifdef HAVE_FSTRM
   if (isEnabledForResponses(fstrmLoggers)) {
-    logFstreamResponse(fstrmLoggers, ip, doTCP, context ? context->d_auth : boost::none, buf, queryTime, *now);
+    logFstreamResponse(fstrmLoggers, ip, doTCP, context ? context->d_auth : boost::none, buf, queryTime, *now, clientip);
```

Comment:
For the UDP case it looks like we might not set the client IP if we don't log queries?

Reply:
Yes, in that case we would log an all-zero clientip... Will look into that.